contracts/VaultV3.vy

```diff
@@ -825,32 +841,34 @@ def _process_report(strategy: address) -> (uint256, uint256):
         self.strategies[strategy].current_debt -= loss
         self.total_debt -= loss
 
-    # Calculate how long until the full amount of shares is unlocked
-    remaining_time: uint256 = 0
     # NOTE: should be precise (no new unlocked shares due to above's burn of shares)
     # newly_locked_shares have already been minted / transfered to the vault, so they need to be substracted
     # no risk of underflow because they have just been minted
     previously_locked_shares: uint256 = self.balance_of[self] - newly_locked_shares
-    _full_profit_unlock_date: uint256 = self.full_profit_unlock_date
-    if _full_profit_unlock_date > block.timestamp: 
-      remaining_time = _full_profit_unlock_date - block.timestamp
 
     # Vault insta unlocks losses and fees to avoid pps decrease
     # NOTE: it can only unlock shares that are previously locked. Any loss / fees over the amount of total locked shares will have an effect on pps
-    shares_to_burn = min(shares_to_burn, previously_locked_shares + newly_locked_shares)
     if shares_to_burn > 0:
+      shares_to_burn = min(shares_to_burn, previously_locked_shares + newly_locked_shares)
       self._burn_shares(shares_to_burn, self)
       # we burn first the newly locked shares, then the previously locked shares
       shares_to_unlock: uint256 = min(shares_to_burn, newly_locked_shares)
       newly_locked_shares -= shares_to_unlock
       previously_locked_shares -= (shares_to_burn - shares_to_unlock)
 
-    # if fees are non-zero, issue shares
-    if protocol_fees > 0:
-      self._issue_shares_for_amount(protocol_fees, protocol_fee_recipient)
+    # issue shares
+    if fees_shares + protocol_fees_shares > 0:
+        self.balance_of[accountant] += fees_shares
+        self.balance_of[protocol_fee_recipient] += protocol_fees_shares
```

Comment:
do you want to break this up into 2 separate checks like the original one was or did you intend to combine them? Shouldnt cause any problems but you may log alot of unnecesary events for the times protocol_fees_shares ==0 due to not being set or being assesed within the PROTOCOL_FEE_ASSESSMENT_PERIOD